ddl/ddl_api.go

```diff
@@ -893,8 +893,18 @@ func checkColumnValueConstraint(col *table.Column, collation string) error {
 	}
 	valueMap := make(map[string]bool, len(col.Elems))
 	ctor := collate.GetCollator(collation)
+	enumlengthlimit := config.GetGlobalConfig().EnableEnumLengthLimit
+	desc, err := charset.GetCharsetDesc(col.Charset)
+	if err != nil {
+		return errors.Trace(err)
+	}
 	for i := range col.Elems {
 		val := string(ctor.Key(col.Elems[i]))
+		if enumlengthlimit {
+			if len(val) > 255 || len(val)*desc.Maxlen > 1020 {
+				return types.ErrTooLongValueInType.GenWithStackByArgs(col.Name)
```

Comment:
We'd better add info like "check enable-enum-length-limit config" to make users skip this error.

Reply:
Done.